sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,103 @@
+from sympy import symbols, Matrix, expand, together
+from sympy.physics.control.lti import TransferFunction, is_proper
+
+a, b, g, s, p = symbols('a, b, g, s, p')
+
+
+def test_TransferFunction_create():
+    G = Matrix([1 / s, 1 / (s + 1)])
+    tf1 = TransferFunction(G)
+    assert tf1.G == G
+
+    G = Matrix([[s / (s**2 + 3), 1 / s],
+            [1 / (s + g), s / (s**2 - g*s - 3)]])
+    tf2 = TransferFunction(G)
+    assert tf2.G == G
+
+    G = Matrix([[(4*p - 10) / (2*p + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    tf3 = TransferFunction(G)
+    assert tf3.G == G
```

Comment:
Same comment I've had on the other PR, I'm worried that if we don't store the numerators and denominators separealy in the class that we will cripple our ability to properly develop some useful methods. The only thing that immediately comes to mind is pole, zero cancelation. If the user creates an expression of ratios of two polynomials, sympy may do some basic simplifications that you don't want to occur. Those simplifications should only occur if the user asks sympy to do them.

My hunch is that we need this api:

```
TransferFunction(s, s**2 + s + 1)
```

This give the most basic SISO transfer function that stores the numerator and denominator separately internally.

If you want a matrix of transfer functions then there are two options:

```python
tf1 = TransferFunction(num, den)
tf2 = TransferFunction(num, den)
TF = TransferFunctionMatrix([tf1], [tf2])
```

or even simply:

```python
tf1 = TransferFunction(num, den)
tf2 = TransferFunction(num, den)
TF = Matrix([tf1], [tf2])
```
For what it is worth, I think making the most basic buidling block be a SISO transfer function makes sense, you can always create classes representing collections of transfer functions.
It isn't clear to me what utility a class that represents a matrix of transfer functions offers that a sympy Matrix of SISO transfer functions can't.

Reply:
Another reason to want a SISO TF as the basic block. You want to be able to do this easily:

```
tf1 = TF(n, d)
tf2 = TF(n, d)
tf3 = TF(n, d)
G = tf1*tf2  # series connection
from control import s
X = Function('X')(s)
Y = F*X
```

Have to worry about matrix shapes in these kinds of operations will be a pain if you are doing SISO work. SISO systems will likely be the primary use case.

Reply:
Sorry for late reply. I've gone through some other resources and I agree with all the things you've said. We should have one SISO transfer function object (although, considering SISO as 1x1 matrix TF was one option that could work).
> If you want a matrix of transfer functions then there are two options:
> ```
> tf1 = TransferFunction(num, den)
> tf2 = TransferFunction(num, den)
> TF = TransferFunctionMatrix([tf1, tf2])
> ```
I think this looks good. I can work on this API, just need some more time. If we decide to have another class named `TransferFunction` (for representing a single rational function), we should define all functionality for that too. It should include addition of two tfs, multiplication, negation, is_proper, is_strictly_proper (for a single tf).

> SISO systems will likely be the primary use case.

If that is the case, should we just work on just SISO instead?

Reply:
I would like you to create a SISO transfer function class first unless there is a compelling reason not to.

Reply:
Okay! I'm working on it.

Reply:
@moorepants if we want to represent a Transfer function as a ratio of polynomials, then it has to somehow use `Mul`.  Or do you just want something like this?
```
>>> tf1 = SISOTransferFunction(s + 1, s**2 + s + 1)
>>> tf1
SISOTransferFunction(s + 1, s**2 + s + 1)
```
I added unit tests in the most recent commit. And the most basic example looks like this:
```
>>> tf1 = SISOTransferFunction(s + 1, s**2 + s + 1)
>>> tf1 
(s + 1)/(s**2 + s + 1)
```

Reply:
> And the most basic example looks like

Python control has a similar kind of outside api...

Reply:
> outside api

What is an "outside api"?

Reply:
> outside api

API*